core/src/main/java/feast/core/job/JobUpdateTask.java

```diff
@@ -103,9 +103,13 @@ public Job call() {
 
   boolean featureSetsChangedFor(Job job) {
     Set<String> existingFeatureSetsPopulatedByJob =
-        job.getFeatureSets().stream().map(FeatureSet::getId).collect(Collectors.toSet());
+        job.getFeatureSets().stream()
+            .map(fs -> fs.getProject() + "/" + fs.getName())
```

Comment:
Seems like it should be encapsulated in a model, not another encoding the programmer needs knowledge of at every use site.